core/shared/src/main/scala/org/http4s/headers/X-Frame-Options.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2013 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s
+package headers
+
+import cats.data.NonEmptyList
+import cats.parse.Parser
+import org.http4s.internal.parsing.CommonRules
+import org.typelevel.ci.CIString
+import org.typelevel.ci.CIStringSyntax
+
+/** The X-Frame-Options HTTP response header can be used to indicate whether or not a browser should be allowed to
+  * render a page in a frame or iframe.
+  *
+  * [[https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options]]
+  *
+  * @param value can be `DENY` or `SAMEORIGIN`.
+  */
+sealed abstract case class `X-Frame-Options` private (value: CIString)
```

Comment:
```suggestion
sealed abstract class `X-Frame-Options` private (val value: CIString)
```

Reply:
@danicheg did you mean to suggest making it `AnyVal` as well? Because otherwise both `hashCode` and `equals` have to be implemented by hand, I think.

Reply:
Oh, I've just realized – it is just two values "deny" and "sameorigin', right?
Then yes, it can be just abstract sealed class of course.. Sorry for the hustle.